setup/symfony_server.rst

```diff
@@ -0,0 +1,271 @@
+internal PHP web server, etc.) However, Symfony provides its own web server to
+make you more productive while developing your apps.
+
+Although this server is not intended for production use, it supports HTTP/2, TLS/SSL,
+automatic generation of security certificates, local domains, and many other
+features that sooner or later you'll need when developing web projects.
+
+Installation
+------------
+
+The Symfony server is distributed as a free installable binary without any
+dependency and support for Linux, macOS and Windows. Go to `symfony.com/download`_
+and follow the instructions for your operating system.
+
+Getting Started
+---------------
+
+The Symfony web server is started once per project, so you may end up with
+several instances (each of them listening to a different port). This is the
+common workflow to serve a Symfony project:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+    $ symfony server:start
+
+      [OK] Web server listening on http://127.0.0.1:....
+      ...
+
+    # Now, browse the given URL in your browser, or run this command:
+    $ symfony open:local
+
+Running the server this way makes it display the log messages in the console, so
+you can't run other commands. If you prefer, you can run the Symfony server in
+the background:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+
+    # start the server in the background
+    $ symfony server:start -d
+
+    # continue working and running other commands...
+
+    # show the latest log messages
+    $ symfony server:log
+
+Enabling TLS
+------------
+
+Browsing the secure version of your apps locally is important to detect
+problems with mixed content early, and to run libraries that only run in HTTPS.
+Traditionally this has been painful and complicated to set up, but the Symfony
+server automates everything. First, run this command:
+
+.. code-block:: terminal
+
+    $ symfony server:ca:install
+
+This command creates a local certificate authority, registers it in your system
+trust store, registers it in Firefox (this is required only for that browser)
+and creates a default certificate for ``localhost`` and ``127.0.0.1``. In other
+words, it does everything for you. You can now browse your local app using
+HTTPS instead of HTTP.
+
+Different PHP Settings Per Project
+----------------------------------
+
+Selecting a Different PHP Version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you have multiple PHP versions installed in your computer, you can tell
+Symfony which one to use creating a file called ``.php-version`` at the project
+root dir:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+
+    # use a specific PHP version
+    $ echo "7.2" > .php-version
+
+    # use any PHP 7.x version available
+    $ echo "7" > .php-version
+
+This other command is useful if you don't remember all the PHP versions
+installed on your computer:
+
+.. code-block:: terminal
+
+    $ symfony local:php:list
```

Comment:
Probably interesting to mention that we list all supported SAPIs,  like CGI, FastCGI, ... which is important as having FastCGI installed is probably the best (as this is what you will have in production -- but not available on Windows). We favor FastCGI (php-fpm) then CGI (which acts as a FastCGI server as well), and finally, we fall back to plain CGI.